src/Components/Search/Previews/Grids/ArtistSearch/RelatedArtworks.tsx

```diff
@@ -7,54 +7,62 @@ import { get } from "Utils/get"
 import { Media } from "Utils/Responsive"
 
 import { RelatedArtworksPreview_viewer } from "__generated__/RelatedArtworksPreview_viewer.graphql"
+import styled from "styled-components"
 import { PreviewGridItemFragmentContainer as PreviewGridItem } from "../PreviewGridItem"
 import { NoResultsPreview } from "./NoResults"
 
 interface RelatedArtworksPreviewProps {
   viewer: RelatedArtworksPreview_viewer
   searchState?: SearchBarState
 }
+
+const ItemContainer = styled(Box)<{ itemsPerRow: 1 | 2 }>`
+  &:nth-child(even) {
+    margin-left: ${p => (p.itemsPerRow === 2 ? space(2) : 0)}px;
+  }
+`
+
 export class RelatedArtworksPreview extends React.Component<
   RelatedArtworksPreviewProps
 > {
   componentDidMount() {
-    const items = get(
-      this.props.viewer,
-      x => x.filter_artworks.artworks_connection.edges,
-      []
-    ).map(x => x.node)
+    const items = this.getArtworks()
 
     this.props.searchState.registerItems(items)
   }
 
-  render() {
-    const { viewer, searchState } = this.props
-
-    const artworks = get(
-      viewer,
+  getArtworks(): any {
```

Comment:
The artworks were being extracted multiple times in the component. This _might_ indicate that something could change with our fragment, but I don't know enough to be sure.

Reply:
Yea, I don't think it's a huge deal to do stuff like `this.props.viewer.edges.map((....` in multiple methods in a component (it's likely multiple components will all need some sort of massaged data like an array of names or what not).

The getter is what I've also generally used to pretty much indicate that (and I'd imagine there's some memoization going on for perf).

Reply:
Oh lol I just realized this isn't actually a getter. This can be `get artworks() { ... }`, and then you could call it as `this.artworks`. I honestly don't know if there's _any_ perf benefit here but overall I think repeatedly munging a small array like this is fine, although any way to avoid it is good!

Reply:
Oh cool, I'll try that.

The main reason I didn't like the repetition of this stuff was that the chain was pretty long - `x.filter_artworks.artworks_connection.edges`. That seems like an easy thing to mess up.

Reply:
All set - take a look and let me know what you think!

Reply:
I really like this extraction, nice one!! 💯